auth-pam.c

```diff
@@ -661,6 +661,18 @@ sshpam_store_conv(int n, sshpam_const struct pam_message **msg,
 
 static struct pam_conv store_conv = { sshpam_store_conv, NULL };
 
+void
+sshpam_cleanup_in_child(void)
+{
+	if (sshpam_handle == NULL)
+		return;
+
+#ifdef PAM_DATA_SILENT
+	/* macOS PAM doesn't support PAM_DATA_SILENT.  */
+	pam_end(sshpam_handle, PAM_SUCCESS | PAM_DATA_SILENT);
```

Comment:
```suggestion
	pam_end(sshpam_handle, sshpam_err | PAM_DATA_SILENT);
```

Shouldn't be still be ended even outside pam-linux?

Reply:
I'm not sure it's safe to do so without `PAM_DATA_SILENT`, otherwise `pam_end` may result in modules trying to perform cleanup work meant for the parent (deleting directories, logging, etc.)

Reply:
If the forked process is going to be run, I would expect `sshpam_err` to always be `PAM_SUCCESS`. When one of the `pam_authenticate()` etc calls fail to return `PAM_SUCCESS`, I would have expected this fork to not have happened.

+1 to @Xyene - this does not change the need for the parent to call `pam_end(sshpam_err)` and that call should not have `PAM_DATA_SILENT` or'd in.